### DISCUSSION TITLE
Parse html with css classes to then produce CSS

### DISCUSSION THREAD

**Discussion by max-ch9i** [tailwindlabs/tailwindcss#9269] [Help]
In my CMS set-up, there exists a component that outputs raw HTML on the page. I want to parse tailwind classes that are supplied with that HTML, in the context of a config file, and add the CSS to the page with the HTML.

The API I'm looking for can be described as follows.
```js
import tw from '@tailwind/hope-to-exist-api';

const html = await fetch(...); // html == `<h1 class="text-slate-300 custom-class">Hello World!</h1>`
const cssOutput = tw.parse(html, `config.tailwind.js`);
// cssOutput == `.text-slate-300 {...} .custom-class {...}`;
appendToDocument(`<style>${cssOutput}</style>` + html);
```

Does TW have such API?

Thanks!
Max

**Comment by wongjn** [comment]
You can use PostCSS with [raw content Tailwind configuration](https://tailwindcss.com/docs/content-configuration#configuring-raw-content) for this:

```js
import postcss from 'postcss';
import tailwindcss from 'tailwindcss';
import baseConfig from './config.tailwind.js';
import sourceCss from './input.css';

const plugins = {
  tailwindcss({
    presets: [baseConfig],
    content: [{ raw: await fetch(...), extension: 'html' }],
  }),
};
// This might not work as-is since I don't work with PostCSS directly,
// but should give you a general idea of usage.
// See https://postcss.org/api for PostCSS documentation.
const { css } = await postcss(plugins).process(sourceCss);
appendToDocument(`<style>${css}</style>` + html);
```

Otherwise, if the above is unfeasible, you could try adding all classes that might exist in the raw HTML into the [safelist configuration](https://tailwindcss.com/docs/content-configuration#safelisting-classes).

**Reply by max-ch9i** [reply]
Thanks, I'll give it a try. Extracted classes will overlap with the ones already produced at the build phase. This will be one challenge.
